app/src/org/commcare/tasks/SaveToDiskTask.java

```diff
@@ -94,10 +94,10 @@ public SaveToDiskTask(int formRecordId, int formDefId, String formRecordPath, Bo
         } catch (XPathException xpe) {
             String cleanedMessage = "An error in your form prevented it from saving: \n" +
                     xpe.getMessage();
-
             return new ResultAndError<>(SaveStatus.SAVE_ERROR, cleanedMessage);
         }
 
+        FormEntryActivity.mFormController.setFormSaveTriggered(true);
```

Comment:
is there a reason we set this here, rather than after the form is fully processed?

Reply:
@ctsims  Thought it will also provide protection from 2 instances of this task running together which potentially can result in 2 saves on the form.

Reply:
Oh, I meant just inserting this same call, but at the bottom of the method, since in theory the form could fail to save below? Or is that unrecoverable after the initial save passes above?

Reply:
Hmm that makes sense. I was more thinking about protecting the form save from two parallel SaveTask threads but looking at the code it doesn't look possible. Made the change.